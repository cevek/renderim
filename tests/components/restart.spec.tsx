import {createElement, ErrorBoundary, getCurrentComponent, restartComponent, ComponentInstance} from 'renderim';
import {render} from '../base';

test('Dedup component restarts', () => {
    let foo!: ComponentInstance;
    let bar!: ComponentInstance;
    function Foo(props: {}) {
        foo = getCurrentComponent();
        return (
            <span>
                <Bar />
            </span>
        );
    }
    function Bar(props: {}) {
        bar = getCurrentComponent();
        return <span>123</span>;
    }

    const result = render(
        <div>
            <Foo />
        </div>,
    );
    restartComponent(foo);
    restartComponent(bar);
    expect(result.tree).toMatchSnapshot('1.root');
    // expect(result.getNextRestartedComponent()).toMatchSnapshot('2.Inner ErrorBoundary');
    // expect(result.getNextRestartedComponent()).toMatchSnapshot('3.Outer ErrorBoundary');
    expect(result.getNextRestartedComponent()).toBeUndefined();
});
