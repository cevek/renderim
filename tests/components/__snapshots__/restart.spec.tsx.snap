// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dedup component restarts: 1.root 1`] = `
"<div>
   <Foo>
      <span>
         <Bar>
            <span>
               123
            </span>
         </Bar>
      </span>
   </Foo>
</div>"
`;
